### PR TITLE
Refactor suse_ha formula

### PR DESCRIPTION
**Pull request by tacerus**
- use defaults map
- move redundant blocks to macros
- allow custom sysconfig values using pillar
- add example pillar
- start resources which are not running
- allow use of custom IPMI ports

### CODE REVIEW

File: suse_ha-formula/pillar.example [hunk 1]
```diff
@@ -15,13 +15,13 @@ suse_ha:
...
-    cluster_secret: !!binary |
-      LGljyn1fBRRcxLxFEgOmhILNTFY/13Cn3EwqqaBN6ynrX6flhiGyTjfW8eAQ1zlJex3uO9kssIcANw9uXLLpOCJ/Fvia3yzHNzCIxfW0zayUOBSMypN1TMKjad5/n8frAFZWNBhTcbk1Cwi764yBj8ErhsPh264qEreRzznJFGI=
     # native Corosync only requires the nodeid for IPv6 based operation - this formula always requires it
     nodeid: 1
   multicast:
     # configure the bind address in a node-specific pillar and have it merged
     bind_address: '192.168.121.55'
     address: '239.0.0.1'
+  # generate an authkey using `corosync-keygen` and transform it to base64 using `base64 -w0 /etc/corosync/authkey`
+  # it is highly recommended to additionally encrypt the base64 string using PGP, in which case the file header needs to be #!gpg|yaml for the binary to stay in tact
```
Comment (cboltz) [inline suse_ha-formula/pillar.example:25]: ´s/in tact/intact/` please

Comment (tacerus) [reply suse_ha-formula/pillar.example]: Thanks!

**Comment by tacerus** [comment]
This should now be ready, tested with pillar:

```
suse_ha:
  resources_dir: /data/resources
  fencing:
    stonith_enabled: true
    ipmi:
      hosts:
        {%- for i in [0, 1] %}
        dev-ipmi{{ i }}:
          ip: 192.168.120.1
          port: 6001{{ i }}
          user: admin
          interface: lanplus
          priv: ADMINISTRATOR
          secret: password
        {%- endfor %}
  cluster:
    name: dev
    ip_version: ipv4
  multicast:
    address: '239.0.0.1'
  cluster_secret: !!binary xxx
  resources:
    cluster-httpd:
      class: ocf
      type: apache
      attributes:
        configfile: /etc/apache2/httpd.conf
      operations:
        monitor:
          interval: 120s
          timeout: 60s
      meta_attributes:
        target-role: Started
```

as well as with `fencing:enable: false` and `fencing:stonith_enable: false`.

The resources example could later go into `pillar.example` as well.

**Comment by cboltz** [comment]
I didn't notice any obvious errors. However, I'm not really familiar with the things this formula configures, therefore I'll let you click the merge button yourself ;-)

**Pull request merged**